File: exercism/typescript/grains/grains.js
export const square = (pow) => {
    if (pow < 1 || pow > 64) {
        throw new Error(`Parameter pow must be in range [1-64]. Actual value: ${pow}`);
    }
    return BigInt(2 ** (pow - 1));
};
export const total = () => {
    let current = 1n;
    let sum = current;
    for (let i = 1; i < 64; i++) {
        current = current * 2n;
        sum += current;
    }
    return sum;
};

File: exercism/java/grains/src/main/java/Grains.java
import java.math.BigInteger;
import java.util.stream.IntStream;

class Grains {

    private void validateInput(final int square) {
        if (square < 1 || square > 64) {
            throw new IllegalArgumentException("square must be between 1 and 64");
        }
    }

    BigInteger grainsOnSquare(final int square) {
        validateInput(square);
        return BigInteger.TWO.pow(square - 1);
    }

    BigInteger grainsOnBoard() {
        return IntStream.range(1, 65).mapToObj(x -> grainsOnSquare(x)).reduce(BigInteger.ZERO, BigInteger::add);
    }

}
